shaccson/src/main/java/UserDao.java

```diff
@@ -0,0 +1,121 @@
+                    user.setPassword(rs.getString("password"));
+                    return user;
+                }
+            }
+        } catch (final SQLException e) {
+            System.err.println("Error while retrieving user by ID: " + e.getMessage());
+            e.printStackTrace(System.err);
+        }
+        return null;
+    }
+
+    public void updateUser(final User user) {
+        try (final Connection conn = DbUtil.getConnection();
+             final PreparedStatement statement = conn.prepareStatement(UPDATE_USER_QUERY)) {
+            statement.setString(1, user.getUsername());
+            statement.setString(2, user.getEmail());
+            statement.setString(3, hashPassword(user.getPassword()));
+            statement.setInt(4, user.getId());
+            final int affectedRows = statement.executeUpdate();
+            if (affectedRows > 0) {
+                System.out.println("User with ID " + user.getId() + " updated successfully.");
+            } else {
+                System.out.println("User update failed. No rows affected.");
+            }
+        } catch (final SQLException e) {
+            System.err.println("Error while updating user: " + e.getMessage());
+            e.printStackTrace(System.err);
+        }
+    }
+
+    public void deleteUser(final int id) {
+        try (final Connection conn = DbUtil.getConnection();
+             final PreparedStatement statement = conn.prepareStatement(DELETE_USER_QUERY)) {
+            statement.setInt(1, id);
+            final int affectedRows = statement.executeUpdate();
+            if (affectedRows > 0) {
+                System.out.println("User with ID %d deleted successfully.".formatted(id));
+            } else {
+                System.out.println("User delete with ID %d failed. No rows affected.".formatted(id));
+            }
+        } catch (final SQLException e) {
+            System.err.println("Error while deleting user: " + e.getMessage());
+            e.printStackTrace(System.err);
+        }
+    }
+
+    public List<User> getAllUsers() {
+        final List<User> users = new ArrayList<>();
+        try (final Connection conn = DbUtil.getConnection();
+             final PreparedStatement statement = conn.prepareStatement(SELECT_ALL_USERS_QUERY);
+             final ResultSet rs = statement.executeQuery()) {
+
+            while (rs.next()) {
+                final User user = new User();
+                user.setId(rs.getInt("id"));
+                user.setUsername(rs.getString("username"));
+                user.setEmail(rs.getString("email"));
+                user.setPassword(rs.getString("password"));
+                users.add(user);
+            }
+        } catch (final SQLException e) {
+            System.err.println("Error while fetching all users: " + e.getMessage());
+            e.printStackTrace(System.err);
+        }
+        return users;
+    }
+
+    private String hashPassword(final String password) {
```

Comment:
IMPROVE: udelal bych public at muzu pouzivat jinde pro hashovani hesla pred ulozenim.

```suggestion
    public String hashPassword(final String password) {
```

Reply:
metoda je nyni public